packages/go_router/lib/src/delegate.dart

```diff
@@ -146,15 +160,36 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
     return true;
   }
 
-  /// Replaces the top-most page of the page stack with the given one.
+  /// Replaces the top-most page of the page stack with the given one. The page
+  /// key of the new page will always be different from the old one.
```

Comment:
```suggestion
  /// Replaces the top-most page of the page stack with the given one.
  /// 
  /// The page key of the new page will always be different from the old one.
```

Reply:
Sorry for that. Hopefully [docs: Improve the documentation](https://github.com/flutter/packages/pull/2846/commits/58853df8c5b0a416f28250b42378932fcfc7721a) addressed all your comments